src/components/Models/MultipleAnimationModel/MultipleAnimationModel.component.tsx

```diff
@@ -64,19 +67,47 @@ export const MultipleAnimationModel: FC<MultipleAnimationModelProps> = ({
     const newClip = loadedAnimations[activeAnimation];
 
     if (!newClip || !mixer) return;
-    if (prevAction && prevAction.getClip().name === newClip.name) return;
+    if (prevAction && prevAction.getClip().name === newClip.name && activeAnimation !== idleAnimation) return;
 
     const newAction = mixer.clipAction(newClip);
-    activeActionRef.current = newAction;
+    if (activeAnimation === idleAnimation) {
+      newAction.setLoop(LoopRepeat, Infinity);
+    } else {
+      newAction.setLoop(LoopOnce, 1);
+      newAction.clampWhenFinished = true;
+    }
+
+    const handleAnimationEnd = (event: { action: AnimationAction }) => {
+      if (event.action === newAction) {
+        onAnimationEnd?.(newAction);
+        if (activeAnimation !== idleAnimation) {
+          const idleClip = loadedAnimations[idleAnimation];
+          if (idleClip) {
+            const idleAction = mixer.clipAction(idleClip);
```

Comment:
Not sure but maybe we should clip all potential actions before and avoid rerunning this where possible. Three js might be funky with the memory.

Reply:
Just a thought on changing animations on handleAnimationEnd. As we are invoking the onAnimationEnd callback, it might be better that we change the animation at top level using setActiveAnimation? Otherwise we then have a disconnect with our activeAnimation prop and the actual animation playing?

Reply:
Yes, exactly, without idleAnimation prop - visage consumer will rely only on `onAnimationEnd` and change activeAnimation to idle

Reply:
Does mean in the /nova example though, going from idle->victory and letting the above code to reset back to idle. We cant activate victory again as `activeAnimation` is already "victory".